tests/catalog/test_rest.py

```diff
@@ -277,36 +277,55 @@ def test_properties_sets_headers(requests_mock: Mocker) -> None:
     )
 
     catalog = RestCatalog(
-        "rest", uri=TEST_URI, warehouse="s3://some-bucket", **{"header.Content-Type": "application/vnd.api+json"}
+        "rest",
+        uri=TEST_URI,
+        warehouse="s3://some-bucket",
+        **{"header.Content-Type": "application/vnd.api+json", "header.Customized-Header": "some/value"},
```

Comment:
Ensure custom header is still supported, meanwhile disallow default headers, such as `Content-Type` to be overwritten.